### PR TITLE
Pressing escape in wunderbar should do something useful

### PR DESCRIPTION
**Pull request by ddifiore**
## Fixes

Issue Number: 2116

## What is the current behavior?

Pressing escape would cause the wunderbar to lose focus.

## What is the new behavior?

Pressing escape now clears out the wunderbar and brings it in focus so the user can continue searching. Also Ctrl+K will now highlight anything in the wunderbar and bring it into focus.

## PR Checklist

<details><summary>Pressing escape in wunderbar should do something useful.</summary>

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [ ] I added a line describing my change to CHANGELOG.md
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

</details>

### CODE REVIEW

File: ui/component/wunderbarSuggestions/view.jsx [hunk 1]
```diff
@@ -228,12 +229,19 @@ export default function WunderBarSuggestions(props: Props) {
     function handleKeyDown(event) {
       const { ctrlKey, metaKey, keyCode } = event;
 
+      if (keyCode === K_KEY_CODE && ctrlKey) {
```
Comment (jessopb) [inline ui/component/wunderbarSuggestions/view.jsx:232]: I'm not certain that the next block `if (!inputRef.current) return;` is necessary, but if it is, I imagine it should be before this one. (Bail early?)

File: ui/component/wunderbarSuggestions/view.jsx [hunk 2]
```diff
@@ -228,12 +229,19 @@ export default function WunderBarSuggestions(props: Props) {
...
+      }
+      
       if (!inputRef.current) {
         return;
       }
 
       if (inputRef.current === document.activeElement && keyCode === ESC_KEY_CODE) {
-        inputRef.current.blur();
+        inputRef.current.value = "";
+        inputRef.current.focus();
```
Comment (jessopb) [inline ui/component/wunderbarSuggestions/view.jsx:244]: Previously, pressing escape would blur the search focus but leave the text.
Now we're clearing the text, but keeping the focus. 

Youtube seems to clear the suggestion box on first ESC, and then blur focus on second ESC press. 
Does this seem like a good idea?

**Review by jessopb — COMMENTED** [review]
This makes sense. But, since users are used to esc clearing focus, I think we need to keep that behavior on second press.

File: ui/component/wunderbarSuggestions/view.jsx (hunk 1 above)
Comment (ddifiore) [reply ui/component/wunderbarSuggestions/view.jsx:232]: You are right!  I thought that line was guarding against when the wunderbar is not selected but it is actually just checking that element exists.  I will make this change.

**Comment by jessopb** [comment]
When you push up the other change, the linter complained about the double-quotes that should be single-quotes, and a trailing space somewhere.  you should be able to run "yarn precommit" to check those

**Comment by tzarebczan** [comment]
Hey @ddifiore , appreciate you taking a look at this issue and welcome to our repo! We'll get another review on this once you let us know it's good to go. 

After that, can we show you some [appreciation](https://lbry.com/faq/appreciation)?

File: ui/component/wunderbarSuggestions/view.jsx (hunk 2 above)
Comment (ddifiore) [reply ui/component/wunderbarSuggestions/view.jsx]: I implemented a hybrid of the two where escape will clear the text and keep the wunderbar in focus if there was any text in the box.  If there is no text in the box then the box will lose focus.  I think this is a good compromise.  It keeps the old behavior if you press escape twice and also keeps the clearing of text for faster searching.

**Pull request merged**